File: WikiMiner - Python/Algorithms-to-Wikification-master/scripts/read20perFold.py
from numpy import *
import os


# implementação do randSample do MatLab
def randSample(arr, perms):
    v = random.permutation(arr)
    r = v[0:perms]
    return r


# end


def read20perFold(idFold, typeSet, perc, out_path):
    filename = out_path + typeSet + '_' + str(idFold) + '.txt'
    fileId = open(filename, 'r')
    In = loadtxt(fileId, delimiter=",", dtype='int')
    In = In[:, 0]  # Pegando apenas os valores de index quando o .txt está no formato "index, id"
    fileId.close()

    k = round(In.shape[0] * perc)
    docsIds = randSample(In, k)

    return docsIds.reshape(-1, 1)
# end

File: utils/matlab-codes-revised/read20perFold.m
% Codes by: Raoni Ferreira.

function [docsIds] = read20perFold(path_output,idFold,typeSet, perc)

filename = strcat(path_output, 'codes-to-create-school\virtual_folds\fixed\',typeSet,'_',num2str(idFold),'.txt');

fileId = fopen(filename,'r');
in = textscan(fileId,'%d','Delimiter','');
fclose(fileId);

docsIds = randsample(in{1},round(size(in{1},1)*perc));

end
